src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -2907,7 +2907,7 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
               // We're looking for a method (as indicated by FUNmode in the silent typed below),
               // so let's make sure our expected type is a MethodType
               val methArgs = NoSymbol.newSyntheticValueParams(argpts map { case NoType => WildcardType case tp => tp })
-              silent(_.typed(meth, mode.forFunMode, MethodType(methArgs, respt))) filter (isMonoContext) map { methTyped =>
+              silent(_.typed(meth.duplicate, mode.forFunMode, MethodType(methArgs, respt))) filter (isMonoContext) map { methTyped =>
```

Comment:
This will replace range positions with offset positions. If we know that only one copy of this tree will end up in the final AST, `duplicateAndKeepPositions` is the right choice.